plugin/db/pg/pg.go

```diff
@@ -715,16 +742,23 @@ func (driver *Driver) FindMigrationHistoryList(ctx context.Context, find *db.Mig
 		if err := driver.updateMigrationHistoryStorageVersion(ctx); err != nil {
 			return nil, err
 		}
-		return util.FindMigrationHistoryList(ctx, query, params, driver, find, baseQuery)
+		return util.FindMigrationHistoryList(ctx, query, params, driver, bytebaseDatabase, find, baseQuery)
 	}
 	return history, err
 }
 
 func (driver *Driver) updateMigrationHistoryStorageVersion(ctx context.Context) error {
-	sqldb, err := driver.GetDbConnection(ctx, "bytebase")
+	var sqldb *sql.DB
+	var err error
+	if driver.strictUseDb {
```

Comment:
We should GetDbConnection only when driver.strictDatabase == "". In other words, when driver.strictDatabase != "", we should never switch connection.